spec/presenters/govspeak_presenter_spec.rb

```diff
@@ -6,73 +6,66 @@
-    end
-  end
-
-  context "when the document has external links and no protocol" do
-    let(:body) { "[External Link without protocol](something.external.uk/url/foo.pdf)" }
-
-    it "does not adds rel='external' to anchor tag" do
-      expect(presented).to eq [
-                                  { content_type: "text/govspeak", content: body },
-                                  { content_type: "text/html",
-                                    content: %(<p><a href="something.external.uk/url/foo.pdf">External Link without protocol</a></p>\n) }]
-    end
-  end
-
-  context "when the document has internal links with a protocol (i.e. HTTP:)" do
-    let(:body) { "[Internal Link with protocol](http://www.gov.uk/url/foo.pdf)" }
-
-    it "does not adds rel='external' to anchor tag" do
-      expect(presented).to eq [
-                                  { content_type: "text/govspeak", content: body },
-                                  { content_type: "text/html",
-                                    content: %(<p><a href="http://www.gov.uk/url/foo.pdf">Internal Link with protocol</a></p>\n) }]
-    end
-  end
-
-  context "when the document has internal links and no protocol" do
-    let(:body) { "[Internal Link without protocol](www.gov.uk/url/foo.pdf)" }
-
-    it "does not adds rel='external' to anchor tag" do
-      expect(presented).to eq [
-                                  { content_type: "text/govspeak", content: body },
-                                  { content_type: "text/html",
-                                    content: %(<p><a href="www.gov.uk/url/foo.pdf">Internal Link without protocol</a></p>\n) }]
+  describe ".presented" do
+    let(:expected) { [{ content_type: "text/govspeak", content: content, }] }
+
+    context "when the document has images as inline attachments" do
+      let(:body)        { "![InlineAttachment:foo.jpg]" }
+      let(:title)       { "Picture of a tasty-looking pizza" }
+      let(:content_id)  { 123 }
+      let(:attachments) {
+        [instance_double(Attachment, url: url, content_id: content_id)]
+      }
+
+      context "there is a matching attachment" do
+        let(:url) { "http://assets.publishing.service.gov.uk/url/foo.jpg" }
+        let(:content) { "[embed:attachments:image:123]" }
+
+        it "replaces InlineAttachment syntax with embed:attachments:image" do
+          expect(presented).to eq expected
+        end
+      end
+
+      context "there is no matching attachment" do
+        let(:url) { "http://assets.publishing.service.gov.uk/url/bar.jpg" }
+        let(:content) { "[embed:attachments:image:foo.jpg]" }
+
+        it "replaces InlineAttachment syntax and references the filename" do
+          expect(presented).to eq expected
+        end
+      end
+    end
+
+    context "when the document has inline attachments" do
+      let(:body)        { "[InlineAttachment:foo.pdf]" }
```

Comment:
Should we have a more realistic Govspeak body text?
The underlying regex is pretty straightforward but it would be good to test how it handles multiple inline attachments embedded in more complex Govspeak.